include/hawkbit-client.h

```diff
@@ -113,4 +113,13 @@ int hawkbit_start_service_sync();
 gboolean hawkbit_progress(const gchar *msg);
 gboolean install_complete_cb(gpointer ptr);
 
+/**
+ * @brief Frees the memory allocated by a RestPayload
+ *
+ * @param[in] payload RestPayload to free
+ */
+void rest_payload_free(RestPayload *payload);
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(RestPayload, rest_payload_free)
```

Comment:
We prepare this but do not make use of it here.

Reply:
That's correct. I think that defining the cleanup function to be used for RestPayload g_autoptr here is cleaner compared to introducing this in the first function refactoring commit where it is actually used.